hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/lock/LockService.java

```diff
@@ -244,4 +244,12 @@ public void provideDynamicMetrics(MetricDescriptor descriptor, MetricsCollection
         }
     }
 
+    @Override
+    public boolean destroyRaftObject(CPGroupId groupId, String name) {
+        boolean result = super.destroyRaftObject(groupId, name);
+        name = withoutDefaultGroupName(name + "@" + groupId.getName());
+        FencedLockProxy proxy = proxies.get(name);
+        proxies.remove(name, proxy);
```

Comment:
Can we simply use `proxies.remove(name);`?

Reply:
I don't think so because `name` is not fully qualified, i.e. `myLock@myGroup`, so in that case (other than `default`) `name=myLock` but the key in `proxies` is actually `myLock@myGroup` hence the ugliness there.

Reply:
Seems, that after
`name = withoutDefaultGroupName(name + "@" + groupId.getName());`
it becomes fully qualified.
Also `proxies` map is used for `proxies.get(name)` and for `proxies.remove(name, proxy)`.
So if you find the `FencedLockProxy` value by the key, it will be removed. And if you don't find the value, it means that there is no such key, and is nothing to remove.

Reply:
Hi @arodionov , I think I misunderstood the initial comment you made. What you just wrote echoes the current logic. 

The `name` passed as argument needs to be fully qualified, hence the bit you quoted -- the `...withoutDefaultGroupName(...)...`

> Also proxies map is used for proxies.get(name) and for proxies.remove(name, proxy).
> So if you find the FencedLockProxy value by the key, it will be removed. And if you don't find the value, it means that there is no such key, and is nothing to remove.

Correct. You could have a `null` check before calling the `remove` but it's surplus.

Logic looks good here unless I am misunderstanding something?

Reply:
I've slightly changed the code to eliminate the confusion between Lock's name and the fully qualified name, and created a Lock with the `UUID + "@myGroup"` name.

<img width="656" alt="image" src="https://github.com/hazelcast/hazelcast/assets/1538227/d05cf64c-afda-44e5-9d95-6fb7d0171684">
So, after the call of `withoutDefaultGroupName`, we have a fully qualified name of the lock. The key with the same name is in the `proxies` map.

![image](https://github.com/hazelcast/hazelcast/assets/1538227/f96fd6b0-b499-455a-89c5-3dc859769422)